discord/bot.go

```diff
@@ -722,7 +722,7 @@ func (bot *Bot) handleMessageCreate(guild *GuildState, s *discordgo.Session, m *
 		//either BOTH the admin/roles are empty, or the user fulfills EITHER perm "bucket"
 		perms := len(guild.PersistentGuildData.AdminUserIDs) == 0 && len(guild.PersistentGuildData.PermissionedRoleIDs) == 0
 		if !perms {
-			perms = guild.HasAdminPermissions(m.Author.ID) || guild.HasRolePermissions(s, m.Author.ID)
+			perms = guild.HasAdminPermissions(s, m.Author.ID) || guild.HasRolePermissions(s, m.Author.ID)
 		}
 		if !perms && g.OwnerID != m.Author.ID {
```

Comment:
So I added this line right here (the `if !perms && g.OwnerID != m.Author.ID`); we already have a copy of the guild object from discordGo, so we can check the ownerID right here. It's slightly less clean than checking the permissions within the function, but it also prevents us from passing the session/guild object around a bunch